src/CouchDB.Driver/Settings/CouchSettings.cs

```diff
@@ -146,6 +156,20 @@ public ICouchConfiguration UseCookieAuthentication(string username, string passw
             return this;
         }
 
+        public ICouchConfiguration UseProxyAuthentication(string username, IReadOnlyCollection<string> roles, string? token = null)
+        {
+            if (string.IsNullOrEmpty(username))
+            {
+                throw new ArgumentNullException(nameof(username));
+            }
+
+            AuthenticationType = AuthenticationType.Proxy;
+            Username = username;
```

Comment:
Why not `?? throw new ArgumentNullException(nameof(username));`

Reply:
What about `""`?

Reply:
What do you mean? I see you added an IF, so it's the same as `??`

Reply:
`??` does not check for empty string.

Reply:
Now I get what you mean, forget my comment. I am tired ahah